### ISSUE TITLE
Custom block becomes obsolete, with the inputs changing their type in the block definition

### ISSUE THREAD

**Issue by steveStevenson** [jmoenig/Snap--Build-Your-Own-Blocks#2016]
All instances of a certain custom block converted into the red "Obsolete!" blocks. When I look at the custom block definition, the types of the block's inputs have switched around (an "upvar" becomes a number, a number becomes any type, an inline command becomes any type, etc. kind of thing). The "obsolete" blocks give an error of "**Error** a custom block definition is missing," and the custom block in the "blocks palette" (the custom "for" block) gives an error of "**TypeError** Cannot read property '0' of undefined."

**Comment by jmoenig** [comment]
This is surely a bug! Can you maybe share the broken project with me, I'd like to investigate. Also, can you remember the exact steps that led to this. Ideally I'd be able to reproduce it myself to fix it. Sorry for the issue and thanks for reporting it!

**Comment by steveStevenson** [comment]
Sure, I can share the project link. It's quite large due to the images, though. I've fixed all the parts of the code that used the block, so it's not actively in the code anymore. But, the block definition is still there. It's the custom "for" block, apart from the more official "for" block I also imported afterwards. The URL is "https://snap.berkeley.edu/snapsource/snap.html#present:Username=mPotasky&ProjectName=nurseryRhyme" The rough steps leading up to the bug where:
1. Created the custom block definition
2. Put the custom block multiple places in multiple sprites, including putting more blocks within the "c-shaped commands" input
3. Ran/tested the project multiple times, where everything worked normally as expected, with no errors
4. Turned off "Retina display support" in the project settings
5. When "Retina display support" was turned off, all instances of the custom "for" block converted into "obsolete" blocks
6. Turned back on "Retina display support," and the "obsolete" blocks still remained
7. All instances of the custom "for" block stayed converted as "obsolete" blocks. I looked a bit into a saved XML file, and (though I may have missed something) it appears that the obsolete blocks are stored plainly as obsolete blocks in the XML data.
Thank you, I really appreciate it. Hopefully me experiencing this bug prevents other people from being effected by it.
